lib/jvm.js

```diff
@@ -348,39 +351,48 @@ class JvmMethodMangler {
     this.originalMethod = null;
     this.newMethod = null;
     this.resolved = null;
+    this.impl = null;
+    this.key = methodId.toString(16);
   }
 
   replace (impl, isInstanceMethod, argTypes, vm, api) {
-    const { method, methodId } = this;
-    const env = vm.getEnv();
-    const thread = api['JavaThread::thread_from_jni_environment'](env.handle);
-    withJvmThread(() => {
-      this.originalMethod = fetchJvmMethod(method);
-      this.newMethod = nativeJvmMethod(method, impl, thread);
-      installJvmMethod(this.newMethod, methodId, thread);
-    });
+    const { key } = this;
+    let mangler = revertManglers.get(key);
+    if (mangler !== undefined) {
+      revertManglers.delete(key);
+      this.method = mangler.method;
+      this.originalMethod = mangler.originalMethod;
+      this.newMethod = mangler.newMethod;
+      this.resolved = mangler.resolved;
+    }
+    this.impl = impl;
+    replaceManglers.set(key, this);
+    if (!manglersScheduled) {
+      Script.nextTick(doManglers, vm);
+      manglersScheduled = true;
+    }
```

Comment:
Would be great to move this into a function called from here and below. E.g. `ensureManglersScheduled()` or something like that.

Reply:
Makes sense! I added the function in next commit